### ISSUE TITLE
LLVM 14 incompatible with MacOS 11?

### ISSUE THREAD

**Issue by aminya** [aminya/setup-cpp#53]
https://github.com/aminya/project_options/runs/6100965907?check_suite_focus=true
```
ld: warning: dylib (/Users/runner/hostedtoolcache/llvm/14.0.0/x64/lib/libc++.dylib) was built for newer macOS version (11.6) than being linked (11.0)
ld: warning: dylib (/Users/runner/hostedtoolcache/llvm/14.0.0/x64/lib/libunwind.dylib) was built for newer macOS version (11.6) than being linked (11.0)
ld: warning: dylib (/Users/runner/hostedtoolcache/llvm/14.0.0/x64/lib/libunwind.dylib) was built for newer macOS version (11.6) than being linked (11.0)
ThinLTO: CMakeFiles/main.dir/Release/src/main/main.cpp.o0: error: Invalid record
LLVM ERROR: Can't load module, abort.
clang-14: error: unable to execute command: Abort trap: 6
clang-14: error: linker command failed due to signal (use -v to see invocation)
```

https://github.com/aminya/project_options/pull/122#issuecomment-1104514183

**Comment by ClausKlein** [comment]
No, it works on my OSX:
```bash
bash-3.2$ cmake -B ../build -G Ninja
-- The CXX compiler identification is Clang 14.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/clausklein/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/local/bin/git (found version "2.36.0") 
-- Version: 7.1.3
-- Build type: 
-- CXX_STANDARD: 11
-- Performing Test has_std_11_flag
-- Performing Test has_std_11_flag - Success
-- Performing Test has_std_0x_flag
-- Performing Test has_std_0x_flag - Success
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS - Success
-- Performing Test FMT_HAS_VARIANT
-- Performing Test FMT_HAS_VARIANT - Success
-- Required features: cxx_variadic_templates
-- Performing Test HAS_NULLPTR_WARNING
-- Performing Test HAS_NULLPTR_WARNING - Success
-- Looking for strtod_l
-- Looking for strtod_l - found
-- CPM: adding package fmt@7.1.3 (7.1.3 at /Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789)
-- cxxopts version 2.2.0
-- The C compiler identification is Clang 14.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/clausklein/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- CPM: adding package cxxopts@2.2.0 (v2.2.0 at /Users/clausklein/.cache/CPM/cxxopts/d5c328043c71a6a5670c290d8646b093756db962)
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/clausklein/cmake/TheLartians/PackageProject/build
bash-3.2$ cmake --build ../build --target all
[11/11] Linking CXX static library _deps/fmt-build/libfmt.a
bash-3.2$ cmake --build ../build --target test
[0/1] Running tests...
Test project /Users/clausklein/cmake/TheLartians/PackageProject/build
    Start 1: test
1/1 Test #1: test .............................   Passed    0.24 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.24 sec
bash-3.2$ git status
Auf Branch master
Ihr Branch ist auf demselben Stand wie 'origin/master'.

nichts zu committen, Arbeitsverzeichnis unverändert
bash-3.2$ 
```

**Comment by aminya** [comment]
Could you build with libc++?

**Comment by aminya** [comment]
The libraries shipped with llvm are built for macOS 11.6
